srcpkgs/synapse/template

```diff
@@ -12,26 +12,47 @@ depends="python3-jsonschema python3-frozendict python3-canonicaljson
  python3-setuptools python3-bleach python3-Jinja2 python3-psycopg2 python3-lxml
  python3-saml2 python3-treq python3-macaroons python3-sortedcontainers
  python3-typing_extensions python3-ijson"
-checkdepends="$depends python3-parameterized"
+checkdepends="$depends python3-parameterized python3-mock"
 short_desc="Matrix reference homeserver"
 maintainer="Joel Beckmeyer <joel@beckmeyer.us>"
 license="Apache-2.0"
 homepage="https://github.com/matrix-org/synapse"
 changelog="https://raw.githubusercontent.com/matrix-org/synapse/develop/CHANGES.md"
 distfiles="https://github.com/matrix-org/synapse/archive/v${version}.tar.gz"
-checksum=182facc856f8d0c0d2d0dd4105fae4f851297167c0ee95c202a3aad9ed86c486
+checksum=3c1b73eb36ec3af00868707b929f9b9d1faae9ccc4022c2c46bdb025ef0ead8e
+conf_files="/etc/synapse/homeserver.yaml /etc/synapse/log.yaml"
 
 system_accounts="synapse"
 synapse_homedir="/var/lib/synapse"
 
 make_dirs="
 	/var/lib/synapse 0700 synapse synapse
-	/etc/synapse 0755 synapse synapse"
+	/var/log/synapse 0755 synapse synapse"
+
+post_configure() {
+	# tweak a few lines in the default configs similar to the `generate_config`
+	# and `generate_log_config` scripts, but commenting out
+	# server_name: "SERVERNAME"
+	# instead of setting it to ensure the user does not start synapse without
+	# setting it
+	vsed -e 's,DATADIR,/var/lib/synapse,' \
+	 -e 's,CONFDIR,/etc/synapse,' \
+	 -e 's,pid_file:,#pid_file:,' \
+	 -e 's,server_name: "SERVERNAME",#server_name: "SERVERNAME",' \
+	 -e 's,SERVERNAME.log.config,log.yaml,' \
+	 -i docs/sample_config.yaml
+	vsed -e 's,/var/log/matrix-synapse,/var/log/synapse,' \
+	 -i docs/sample_log_config.yaml
```

Comment:
Would it be too hard to make this a patch instead of vseds? Makes things more explicit, I think.

(See `gitea`).

Reply:
Changed to patch, thanks for the advice!